applications/openshift/etcd/etcd_key_file/rule.yml

```diff
@@ -33,3 +33,19 @@ ocil: |-
     Run the following command on the master node(s):
     <pre>$ grep ETCD_KEY_FILE=/etc/etcd/etcd.conf</pre>
     Verify that there is a key file configured.
+
+{{%- if product == "ocp4" %}}
+warnings:
+    - general: |-
+        {{{ openshift_cluster_setting("/api/v1/namespaces/openshift-etcd/configmaps/etcd-pod") | indent(8) }}}
+{{%- endif %}}
+
+template:
+    name: yamlfile_value
+    vars:
+        ocp_data: "true"
+        filepath: /api/v1/namespaces/openshift-etcd/configmaps/etcd-pod
+        yamlpath: ".data['pod.yaml']"
+        values:
+          - value: ".*--key-file=/etc/kubernetes/static-pod-certs/secrets/etcd-all-serving/etcd-serving-NODE_NAME.key\\.*"
```

Comment:
Does `NODE_NAME` actually get changed to the actual node's hostname?

Reply:
It doesn't. For this check, it should be enough to ensure that --key-file exists in the data and is set to _something_, for etcd this is a required configuration flag when enabling TLS as a whole.

Reply:
Probably should `.*` the hostname then like: `etcd-serving-.*\.key`

Reply:
@JAORMX is `NODE_NAME` literal here?

Reply:
yes